src/test/java/io/cdap/plugin/salesforce/etl/SalesforceBatchMultiSourceETLTest.java

```diff
@@ -163,4 +167,60 @@ public void testBlackList() throws Exception {
     Assert.assertEquals(expectedSObjects, actualSObjects);
   }
 
+  @Test
+  public void testValidateSObjects() throws Exception {
+
+     SalesforceMultiSourceConfig salesforceMultiSourceConfig = new SalesforceMultiSourceConfigBuilder()
+       .setConsumerKey(CONSUMER_KEY).setConsumerSecret(CONSUMER_SECRET).setUsername(USERNAME).setPassword(PASSWORD)
+       .setSecurityToken(SECURITY_TOKEN).setLoginUrl(LOGIN_URL).setWhiteList("Account,Contact").
+       setBlackList("Contact").build();
+
+    List<String> sObjects = getSObjects();
+    MockFailureCollector collector = new MockFailureCollector();
+
+    salesforceMultiSourceConfig.validateSObjects(collector);
+
+    List<String> invalidWhiteListedSObject = salesforceMultiSourceConfig.getWhiteList().stream().filter(name ->
+       !sObjects.contains(name)).collect(Collectors.toList());
+    List<String> invalidBlackListSObject = salesforceMultiSourceConfig.getWhiteList().stream().filter(name ->
+      !sObjects.contains(name)).collect(Collectors.toList());
```

Comment:
Take a look at this test which is also testing a similar validate method - https://github.com/data-integrations/salesforce/blob/develop/src/test/java/io/cdap/plugin/salesforce/SalesforceSchemaUtilTest.java#L159.
You only need to check if the failure collector has errors or not. You shouldn't be calling Salesforce to get the list of sobjects which means you don't need `getSObjects()` and the lines that call it.

Reply:
did changes now we are not calling getSObjects().